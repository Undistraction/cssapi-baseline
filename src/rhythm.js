import { CONFIG } from './const';
import { defaultConfig } from './utils';
import { throwError, invalidConfigMessage } from './errors';
import validateConfig from './validators/validateConfig';

const build = config => `xx`;

const configure = (config = defaultConfig(CONFIG)) =>
  validateConfig(config).matchWith({
    Success: build,
    Failure: ({ value }) => {
      throwError(invalidConfigMessage(value));
    },
  });

export default {
  configure,
};
