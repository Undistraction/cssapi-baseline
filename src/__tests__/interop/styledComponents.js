import React from 'react';
import styled from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import rhythm from '../../index';

describe(`works with styled-components`, () => {
  const baseline = rhythm.configure();
  const Element = styled.div`
    ${baseline(16)};
  `;
  const tree = renderer.create(<Element />).toJSON();

  it(`renders component with correct 'font-size' and 'line-height'`, () => {
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule(`font-size`, `1rem`);
    expect(tree).toHaveStyleRule(`line-height`, `1.25rem`);
  });
});
