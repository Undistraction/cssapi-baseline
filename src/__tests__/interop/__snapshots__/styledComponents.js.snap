// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with styled-components renders component with correct 'font-size' and 'line-height' 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.25rem;
}

<div
  className="c0"
/>
`;
